src/ensemble/BoxDomain.cpp

```diff
@@ -8,65 +8,85 @@
 using namespace std;
 using Log::global_log;
 
+enum BoxConfig
+{
+	BLANK = 0,
+	L = 1, //001b
+	BOXMIN = 2, //010b
+	L_BOXMIN = 3, //011b
+	BOXMAX = 4, //100b
+	L_BOXMAX = 5, //101b
+	BOXMIN_BOXMAX = 6, //110b
+	L_BOXMIN_BOXMAX = 7 //111b
+};
+
 BoxDomain::BoxDomain() {
 	for( int d = 0; d < 3; d++)
 		_rmin[d] = 0;
 }
 
 void BoxDomain::readXML(XMLfileUnits& xmlconfig) {
-	double l[3];
+	std::array<double, 3> l;
+	int tempConfig = 0;
+
 	bool l1, l2, l3, lExists;
 	l1 = xmlconfig.getNodeValueReduced("lx",l[0]);
 	l2 = xmlconfig.getNodeValueReduced("ly",l[1]);
 	l3 = xmlconfig.getNodeValueReduced("lz",l[2]);
 	lExists = l1 & l2 & l3;
-	bool boxMinExists, boxMaxExists;
-	if(boxMinExists = xmlconfig.changecurrentnode("boxMin"))
+	if(lExists)
+		tempConfig |= 1;
+
+	if(xmlconfig.changecurrentnode("boxMin"))
 	{
 		xmlconfig.getNodeValueReduced("x", _rmin[0]);
 		xmlconfig.getNodeValueReduced("y", _rmin[1]);
 		xmlconfig.getNodeValueReduced("z", _rmin[2]);
 		xmlconfig.changecurrentnode("..");
+		tempConfig |= 2;
 	}
-	if(boxMaxExists = xmlconfig.changecurrentnode("boxMax"))
+	if(xmlconfig.changecurrentnode("boxMax"))
 	{
 		xmlconfig.getNodeValueReduced("x", _rmax[0]);
 		xmlconfig.getNodeValueReduced("y", _rmax[1]);
 		xmlconfig.getNodeValueReduced("z", _rmax[2]);
 		xmlconfig.changecurrentnode("..");
+		tempConfig |= 4;
 	}
 
-	//calculations for missing elements
-	//earlier implementation did not check for existence of l values, but this implementation is checking for it now
-	if(!lExists && !boxMinExists && !boxMaxExists)
-	{
-		global_log->error() << "Domain specification defines neither bounds not lengths" << endl;
-		Simulation::exit(1);
-	}
-	else if(lExists && !boxMinExists && !boxMaxExists) //old behavior
-	{
-		for(int d = 0; d < 3; d++)
-		{
-			_rmin[d] = 0;
-			_rmax[d] = l[d];
-		}
-	}
-	else if(boxMinExists && lExists) //does not matter if boxMax exists, overwrite anyway
+	BoxConfig boxConfig = static_cast<BoxConfig>(tempConfig); //value will be within range, so not undefined
+	switch(boxConfig)
 	{
-		for(int d = 0; d < 3; d++)
-			_rmax[d] = _rmin[d] + l[d];
-	}
-	else if(boxMaxExists && lExists) //'all existing' case covered by earlier case
-	{
-		for(int d = 0; d < 3; d++)
-			_rmin[d] = _rmax[d] - l[d];
-	}
-	else if(boxMaxExists && boxMinExists) //do nothing
-	{}
-	else //only boxmin or boxmax exists, overwrite
-	{
-		global_log->error() << "Domain specification does not have enough info to define box dimensions" << endl;
-		Simulation::exit(1);
+		case BLANK:
+			global_log->error() << "Domain specification does not have enough info to define box dimensions. Please make sure the domain specification is correct." << endl;
+			Simulation::exit(1);
+		
+		case L:					// old behaviour
+			for(int d = 0; d < 3; d++)
+			{
+				_rmin[d] = 0;
+				_rmax[d] = l[d];
+			}
+			break;
+		
+		case L_BOXMIN:
+		case L_BOXMIN_BOXMAX: 	// overwriting _rmax
```

Comment:
The issue with this class is that this feature is not implemented; in the background the other classes (like GridFiller etc) still only access the Domain length and assume the origin is at 0,0,0. The rmin and rmax are usually ignored. I'll have to finish implementing the arbitrary origin but it's currently not a priority. Meanwhile, do I leave this class as is? Or do I add some error message?

Reply:
Either implement a working hot fix and throw exceptions for corner cases that are not covered, or throw warnings all the time or fix it properly ;-)
In all cases that do not fully resolve the problem create an issue on GitHub and put the link to it in as a comment in the code.

Reply:
I"ll rollback the file, and add an issue